From d6df537fe5098c0e9f50deecab85b2102049030a Mon Sep 17 00:00:00 2001
From: Stefano Stabellini <sstabellini@kernel.org>
Date: Mon, 21 Sep 2020 13:15:53 -0700
Subject: [PATCH 2/2] don't call busybox --install

Busybox --install is very convenient but has the problem that it only
works for native builds. If busybox is of a different arch compared to
the target, it cannot be run at build time.

Solve the problem by install busybox, and creating links to it, by hand.

Signed-off-by: Stefano Stabellini <stefano.stabellini@xilinx.com>
---
 initrd/applet-list | 308 +++++++++++++++++++++++++++++++++++++++++++++
 initrd/make-initrd |   5 +-
 2 files changed, 312 insertions(+), 1 deletion(-)
 create mode 100644 initrd/applet-list

Index: git/initrd/applet-list
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ git/initrd/applet-list	2020-09-21 18:29:35.560281087 -0700
@@ -0,0 +1,308 @@
+ipneigh
+tee
+mpstat
+lzop
+ip
+inotifyd
+rdate
+nslookup
+switch_root
+sendmail
+slattach
+sum
+od
+less
+vlock
+cmp
+dnsdomainname
+renice
+fsync
+unlzop
+xxd
+stat
+flock
+fbset
+bzip2
+fdisk
+fallocate
+deallocvt
+cat
+mkdir
+traceroute
+dc
+loadfont
+nohup
+wc
+ping6
+login
+ifdown
+shred
+false
+pwd
+uname
+hd
+hdparm
+expand
+lspci
+shuf
+time
+tunctl
+sha3sum
+sed
+ed
+makemime
+factor
+ls
+volname
+stty
+truncate
+nproc
+openvt
+hexdump
+mountpoint
+nl
+more
+wget
+traceroute6
+linuxrc
+fbsplash
+pstree
+kill
+df
+uptime
+split
+arch
+getty
+syslogd
+cpio
+unexpand
+sha512sum
+sha256sum
+clear
+rdev
+md5sum
+egrep
+lzma
+rmdir
+tail
+netstat
+mkswap
+pkill
+fstrim
+bzcat
+chmod
+mount
+paste
+rev
+fsck
+ping
+ifup
+mesg
+gzip
+setconsole
+printenv
+swapoff
+unxz
+sort
+chpasswd
+true
+linux32
+blkid
+env
+dd
+fatattr
+bc
+[[
+mkfifo
+route
+xz
+setkeycodes
+install
+mkpasswd
+awk
+rm
+nsenter
+which
+sh
+ln
+fdflush
+test
+ttysize
+fuser
+base64
+run-parts
+ntpd
+mv
+printf
+nanddump
+zcat
+basename
+xzcat
+unlzma
+ipcs
+ash
+tar
+iplink
+sysctl
+crontab
+chown
+watchdog
+partprobe
+ifconfig
+chvt
+readahead
+add-shell
+ipaddr
+link
+getopt
+reboot
+blkdiscard
+pscan
+nandwrite
+cksum
+iostat
+deluser
+du
+tty
+iprule
+setserial
+eject
+unix2dos
+free
+conspy
+pivot_root
+setfont
+kbd_mode
+strings
+groups
+ionice
+swapon
+smemcap
+brctl
+yes
+cal
+bbconfig
+uudecode
+logger
+watch
+[
+udhcpc
+acpid
+echo
+setsid
+expr
+readlink
+linux64
+sync
+uuencode
+insmod
+rmmod
+raidautorun
+nbd-client
+ipcrm
+modinfo
+adjtimex
+unzip
+cp
+killall
+delgroup
+blockdev
+halt
+modprobe
+showkey
+setlogcons
+whois
+lsof
+vi
+grep
+sha1sum
+ps
+timeout
+crond
+hwclock
+unlink
+touch
+losetup
+iptunnel
+fold
+id
+chroot
+rfkill
+mdev
+lzcat
+microcom
+chgrp
+date
+udhcpc6
+init
+su
+head
+remove-shell
+reset
+realpath
+xargs
+cut
+poweroff
+killall5
+addgroup
+logread
+resize
+pipe_progress
+findfs
+dumpkmap
+reformime
+fdformat
+arping
+nologin
+lsmod
+mkdosfs
+klogd
+dirname
+umount
+setpriv
+cryptpw
+usleep
+pgrep
+hostname
+powertop
+dos2unix
+iproute
+ether-wake
+nice
+vconfig
+gunzip
+hostid
+top
+tr
+sleep
+mkfs.vfat
+comm
+lzopcat
+depmod
+arp
+mktemp
+uniq
+dumpleases
+adduser
+mknod
+dmesg
+nmeter
+lsusb
+find
+unshare
+nameif
+whoami
+ifenslave
+pwdx
+loadkmap
+pmap
+ipcalc
+tac
+bunzip2
+beep
+pidof
+nc
+fgrep
+readprofile
+seq
+diff
Index: git/initrd/make-initrd
===================================================================
--- git.orig/initrd/make-initrd	2020-09-21 18:29:35.564281091 -0700
+++ git/initrd/make-initrd	2020-09-21 18:30:42.528356223 -0700
@@ -58,7 +58,11 @@
     if [ -n "$QEMU_USER" ]; then
         $QEMU_USER $tmpdir/bin/busybox --install $tmpdir/bin
     else
-        echo "QEMU_USER is not defined, no binary symlinks will be available"
+        echo "QEMU_USER is not defined, default binary symlinks will be available"
+        for i in `cat "$base"/applet-list`
+        do
+            ln -s /bin/busybox $tmpdir/bin/$i
+        done
     fi
 else
     $tmpdir/bin/busybox --install $tmpdir/bin
